src/main/java/org/opensearch/plugin/insights/core/service/categorizer/IndicesFieldTypeCache.java

```diff
@@ -0,0 +1,77 @@
+package org.opensearch.plugin.insights.core.service.categorizer;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.RamUsageEstimator;
+import org.opensearch.common.cache.Cache;
+import org.opensearch.common.cache.CacheBuilder;
+import org.opensearch.common.metrics.CounterMetric;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.common.unit.ByteSizeValue;
+import org.opensearch.core.index.Index;
+
+public class IndicesFieldTypeCache {
+
+    private static final Logger logger = LogManager.getLogger(IndicesFieldTypeCache.class);
+    public static final Setting<ByteSizeValue> INDICES_FIELD_TYPE_CACHE_SIZE_KEY = Setting.memorySizeSetting(
+        "search.insights.indices.fieldtype.cache.size",
+        new ByteSizeValue(-1),
+        Setting.Property.NodeScope
```

Comment:
Will need to add a settings update consumer for these and method to validate and set?

Reply:
There is no point of adding settings update consumer, as the setting is not dynamic. The cache size cannot be changed dynamically